### ISSUE TITLE
[MNG-7127] NullPointerException in MavenCliTest.testStyleColors in JDK 16

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#8325]
**[Carlos Chacin](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=cchacin)** opened **[MNG-7127](https://issues.apache.org/jira/browse/MNG-7127?redirect=false)** and commented

## Given

That I'm using JDK 16 to build https://github.com/apache/maven

## When

Executing the command to build the project locally after the git clone

```java
mvn clean verify
```

## Then

There is a failed test

```java
[ERROR] Tests run: 14, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.929 s <<< FAILURE! - in org.apache.maven.cli.MavenCliTest
[ERROR] testStyleColors  Time elapsed: 0.03 s  <<< ERROR!
java.lang.NullPointerException
	at org.apache.maven.cli.MavenCliTest.testStyleColors(MavenCliTest.java:327)
```

## Possible cause:

In JDK 8 the method Paths.get used in [ResolveFile.java#L47](https://github.com/apache/maven/blob/master/maven-embedder/src/main/java/org/apache/maven/cli/ResolveFile.java#L47) is using internally:

```java
public static Path get(String first, String... more) {
    return FileSystems.getDefault().getPath(first, more);
}
```

but in JDK 16

```java
public static Path get(String first, String... more) {
    return Path.of(first, more);
}
```

---

**Issue Links:**
- [MNG-7054](https://issues.apache.org/jira/browse/MNG-7054) [Test] MavenCliTest.testStyleColors failed with Java 16 and 17
 (_**"is duplicated by"**_)
- [MNG-7124](https://issues.apache.org/jira/browse/MNG-7124) java.lang.NullPointerException at org.apache.maven.cli.MavenCliTest.testStyleColors(MavenCliTest.java:327)
 (_**"is duplicated by"**_)

**Remote Links:**
- [GitHub Pull Request #460
](https://github.com/apache/maven/pull/460)

**Backported to:** [4.0.0-alpha-2](https://github.com/apache/maven/milestone/113?closed=1), [3.8.2](https://github.com/apache/maven/milestone/103?closed=1)

**Comment by jira-importer** [comment]
**[Carlos Chacin](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=cchacin)** commented

PR: https://github.com/apache/maven/pull/460

**Comment by jira-importer** [comment]
**[Martin Kanters](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=martinkanters)** commented

Thanks! I've merged it.

I've verified the build failed with jdk 16, then verified that the PR fixed it and lastly verified that the PR won't break the java 8 build.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Fixed with [276c6a8dc445cbaffb2bc2a6344f54abeb9b4311](https://gitbox.apache.org/repos/asf?p=maven.git;a=commit;h=276c6a8dc445cbaffb2bc2a6344f54abeb9b4311).
